service/src/main/java/bio/terra/workspace/service/workspace/flight/CreateGcpContextFlight.java

```diff
@@ -39,6 +39,13 @@ public CreateGcpContextFlight(FlightMap inputParameters, Object applicationConte
 
     RetryRule retryRule = RetryRules.shortExponential();
 
+    addStep(
```

Comment:
This step needs to be first in this flight, before `GenerateProjectIdStep` and `PullProjectFromPoolStep` up above. 
Otherwise, unauthorized users will be able to churn projects by repeatedly calling this endpoint, which could exhaust project creation quota.